### PR TITLE
chore(server): remove token when logged out

### PR DESCRIPTION
**Pull request by alextran1502**

### CODE REVIEW

File: mobile/lib/modules/login/providers/authentication.provider.dart [hunk 1]
```diff
@@ -93,6 +93,7 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
   Future<bool> logout() async {
     state = state.copyWith(isAuthenticated: false);
```
Comment (bo0tzz) [inline mobile/lib/modules/login/providers/authentication.provider.dart:94]: This state change happens before all the cache invalidation etc. If something goes wrong inbetween, eg the logout api call crashes things, will that cause a problem with stale cache?

File: server/apps/immich/src/controllers/auth.controller.ts [hunk 2]
```diff
@@ -59,13 +59,19 @@ export class AuthController {
...
 
+  @Authenticated()
   @Post('logout')
-  async logout(@Req() req: Request, @Res({ passthrough: true }) res: Response): Promise<LogoutResponseDto> {
+  async logout(
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+    @GetAuthUser() authUser: AuthUserDto,
+  ): Promise<LogoutResponseDto> {
+    const token = req.headers.authorization?.split(' ')[1] || req.cookies[IMMICH_ACCESS_COOKIE];
```
Comment (jrasm91) [inline server/apps/immich/src/controllers/auth.controller.ts:69]: Is this still being used?

File: server/apps/immich/src/controllers/auth.controller.ts [hunk 3]
```diff
@@ -59,13 +59,19 @@ export class AuthController {
...
     const authType: AuthType = req.cookies[IMMICH_AUTH_TYPE_COOKIE];
 
     res.clearCookie(IMMICH_ACCESS_COOKIE);
     res.clearCookie(IMMICH_AUTH_TYPE_COOKIE);
 
-    return this.authService.logout(authType);
+    return this.authService.logout(authType, authUser);
```
Comment (jrasm91) [inline server/apps/immich/src/controllers/auth.controller.ts:75]: In general all the services usually have auth user dto as the first argument.

File: server/libs/domain/src/auth/auth.service.ts [hunk 4]
```diff
@@ -76,7 +76,11 @@ export class AuthService {
     return this.authCore.createLoginResponse(user, AuthType.PASSWORD, isSecure);
   }
 
-  public async logout(authType: AuthType): Promise<LogoutResponseDto> {
+  public async logout(authType: AuthType, authUser: AuthUserDto): Promise<LogoutResponseDto> {
+    if (authUser.accessTokenId) {
+      await this.userTokenCore.deleteToken(authUser.accessTokenId);
```
Comment (jrasm91) [inline server/libs/domain/src/auth/auth.service.ts:81]: Looks good!

**Review by jrasm91 — APPROVED** [review]

**Pull request merged**